assets/activation-scripts/activate

```diff
@@ -15,6 +15,69 @@ export _tcsh_home="@out@/activate.d/tcsh_home"
 export _flox_activate_tracelevel="${_FLOX_PKGDB_VERBOSITY:-0}"
 [ "$_flox_activate_tracelevel" -eq 0 ] || set -x
 
+NOT_READY="SOCKET_NOT_READY"
+
+poll_services_status () {
+  local socket_file="$1"
+  local output
+  output=$(process-compose process list -u "$socket_file" -o json 2>&1)
+  # Save whatever the current `pipefail` behavior is so it can be restored
+  local saved_options
+  saved_options=$(set +o)
+  # We don't want to exit on pipe failures here
+  set +o pipefail
+  local parsed_json
+  parsed_json=$(echo "$output" | jq -r -c '.[0].status' 2>/dev/null)
+  # Restore the previous shell settings
+  eval "$saved_options"
+  # `parsed_json` will be a null string if anything went wrong
+  echo "${parsed_json:-${NOT_READY}}"
+}
+
+wait_for_services_socket () {
+  local socket_file="$1"
+  local status
+  status=$(poll_services_status "$socket_file")
+  while [ "$status" == "$NOT_READY" ]; do
+    sleep 0.1
```

Comment:
nonblocking note: I'm curious how long startup should usually take and how often we get to the first timeout. If we're adding almost .1 of unnecessary lag to activations, that's a bit of a bummer

Reply:
That's why I check the status first, so that we can skip the wait if possible. Note that this is also only happening for an activation that starts services. All other activations will be as fast as they were before.

Startup is usually pretty fast since it creates the socket right as it creates the HTTP server. I'm not sure how bummed I am considering that at some point we'll be blocking on health checks, and that could easily take longer than 100ms.

Reply:
If startup takes .00001 longer than checking status though, we'll wait an extra .1

I only think that's significant if startup is usually very fast but slightly slower than checking status

Reply:
Do you have an alternative suggestion?

Reply:
I don't think it matters at this point, but we could investigate the start time a few times, and if it's significantly lower than .1 but we're still sleeping .1, we could start with a smaller timeout and use exponential backoff

Reply:
Or just use a lower interval, idk